core/capabilities/ccip/configs/solana/chain_writer.go

```diff
@@ -0,0 +1,330 @@
+
+const (
+	destChainSelectorPath = "Message.Header.DestChainSelector"
+	destTokenAddress      = "Message.TokenAmounts.DestTokenAddress"
+)
+
+func getCommitMethodConfig(fromAddress string, routerProgramAddress string, sysvarInstructionsAddress string, computeBudgetProgramAddress string, commonAddressesLookupTable solana.PublicKey, routerAccountConfig chainwriter.PDALookups) chainwriter.MethodConfig {
+	return chainwriter.MethodConfig{
+		FromAddress:        fromAddress,
+		InputModifications: nil,
+		ChainSpecificName:  "commit",
+		LookupTables: chainwriter.LookupTables{
+			StaticLookupTables: []solana.PublicKey{
+				commonAddressesLookupTable,
+			},
+		},
+		Accounts: []chainwriter.Lookup{
+			routerAccountConfig,
+			chainwriter.PDALookups{
+				Name: "SourceChainState",
+				PublicKey: chainwriter.AccountConstant{
+					Address: routerProgramAddress,
+				},
+				Seeds: []chainwriter.Seed{
+					{Static: []byte("source_chain_state")},
+					{Dynamic: chainwriter.AccountLookup{Location: "MerkleRoot.DestChainSelector"}},
+				},
+				IsSigner:   false,
+				IsWritable: false,
+			},
+			chainwriter.AccountConstant{
+				Name:       "RouterProgram",
+				Address:    routerProgramAddress,
+				IsSigner:   false,
+				IsWritable: false,
+			},
+			chainwriter.PDALookups{
+				Name: "RouterAccountConfig",
+				PublicKey: chainwriter.AccountConstant{
+					Address:    routerProgramAddress,
+					IsSigner:   false,
+					IsWritable: false,
+				},
+				Seeds: []chainwriter.Seed{
+					{Static: []byte("config")},
+				},
+				IsSigner:   false,
+				IsWritable: false,
+			},
+			chainwriter.PDALookups{
+				Name: "RouterAccountState",
+				PublicKey: chainwriter.AccountConstant{
+					Address:    routerProgramAddress,
+					IsSigner:   false,
+					IsWritable: false,
+				},
+				Seeds: []chainwriter.Seed{
+					{Static: []byte("state")},
+				},
+				IsSigner:   false,
+				IsWritable: false,
+			},
+			chainwriter.PDALookups{
+				Name: "RouterReportAccount",
+				PublicKey: chainwriter.AccountConstant{
+					Address:    routerProgramAddress,
+					IsSigner:   false,
+					IsWritable: false,
+				},
+				Seeds: []chainwriter.Seed{
+					{Dynamic: chainwriter.AccountLookup{
+						Location: "args.MerkleRoots",
+					}},
+				},
+				IsSigner:   false,
+				IsWritable: false,
+			},
+			chainwriter.AccountConstant{
+				Name:       "ComputeBudgetProgram",
+				Address:    computeBudgetProgramAddress,
+				IsSigner:   true,
+				IsWritable: false,
+			},
+			chainwriter.AccountConstant{
+				Name:       "SysvarInstructions",
+				Address:    sysvarInstructionsAddress,
+				IsSigner:   true,
+				IsWritable: false,
+			},
+		},
+		DebugIDLocation: "",
+	}
+}
+
+func getExecuteProgramConfig(fromAddress string, routerProgramAddress string, sysvarInstructionsAddress string, computeBudgetProgramAddress string, commonAddressesLookupTable solana.PublicKey, routerAccountConfig chainwriter.PDALookups) chainwriter.MethodConfig {
```

Comment:
nit: think this name would be more accurate and line up with the commit config
```suggestion
func getExecuteMethodConfig(fromAddress string, routerProgramAddress string, sysvarInstructionsAddress string, computeBudgetProgramAddress string, commonAddressesLookupTable solana.PublicKey, routerAccountConfig chainwriter.PDALookups) chainwriter.MethodConfig {
```